### PR TITLE
Feature/82: Added AWS ELB App-Tier Security plugin and test cases

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

**Review by dipsubha06 — APPROVED** [review]

**Review by giorod3 — COMMENTED** [review]
lets talk about this in the morning

File: plugins/aws/elb/appTierElbSecurity.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    more_info: 'AWS App-Tier ELBs should use the latest predefined security policies to secure the connection between client and ELB.',
+    link: 'https://docs.aws.amazon.com/elasticloadbalancing/latest/application/create-https-listener.html',
+    recommended_action: 'Update App-Tier ELB reference security policy to latest predefined security policy to secure the connection between client and ELB',
+    apis: ['ELB:describeLoadBalancers', 'ELB:describeLoadBalancerPolicies', 'ELB:describeTags', 'STS:getCallerIdentity'],
+    settings: {
+        elb_app_tier_tag_key: {
+            name: 'App-Tier Tag Key',
+            description: 'App-Tier tag key used by ELBs to indicate App-Tier groups',
+            regex: '^.*$',
+            default: 'app-tier'
```
Comment (giorod3) [inline plugins/aws/elb/appTierElbSecurity.js:17]:
```suggestion
            default: ''
```

**Review by giorod3 — COMMENTED** [review]
explain why collectors/aws/elb/describeLoadBalancerPolicies.js was removed and in the collector the override was removed and replaced with filter key and filter value

File: plugins/aws/elb/appTierElbSecurity.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                            region, resource);
+                        return cb();
+                    }
+
+                    var insecurePolicy = false;
+                    var securityPolicyFound = false;
+                    describeLoadBalancerPolicies.data.PolicyDescriptions.forEach(function(policyDesc) {
+                        if(policyDesc.PolicyAttributeDescriptions) {
+                            for (var policyAttrDesc of policyDesc.PolicyAttributeDescriptions) {
+                                if (policyAttrDesc.AttributeName === 'Reference-Security-Policy') {
```
Comment (giorod3) [inline plugins/aws/elb/appTierElbSecurity.js:116]:
```suggestion
                                if (policyAttrDesc.AttributeName && policyAttrDesc.AttributeName === 'Reference-Security-Policy') {
```

File: plugins/aws/elb/appTierElbSecurity.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                        region, resource);
+                    return cb();
+                }
+
+                var appTierTag = false;
+                describeTags.data.TagDescriptions.forEach(function(Tags) {
+                    if(Tags && Tags.Tags) {
+                        for (var i in Tags.Tags){
+                            var td = Tags.Tags[i];
+                            if(td.Key === config.elb_app_tier_tag_key) {
```
Comment (giorod3) [inline plugins/aws/elb/appTierElbSecurity.js:83]:
```suggestion
                            if(td.Key && td.Key === config.elb_app_tier_tag_key) {
```

**Review by giorod3 — APPROVED** [review]

**Pull request merged**